subsys/net/lib/dns/mdns_responder.c

```diff
@@ -720,6 +1200,72 @@ static int register_dispatcher(struct mdns_responder_context *ctx,
 	return dns_dispatcher_register(&ctx->dispatcher);
 }
 
+#if defined(CONFIG_MDNS_RESPONDER_PROBE)
+/* Initialize things so that we can do probing before staring the listener */
+static int pre_init_listener(void)
+{
+	int ok = 0;
+	struct net_if *iface;
+	int iface_count;
+
+	NET_IFACE_COUNT(&iface_count);
+	NET_DBG("Pre-init %s for %d interface%s", "mDNS", iface_count,
+		iface_count > 1 ? "s" : "");
+
+#if defined(CONFIG_NET_IPV6)
+	if ((iface_count > MAX_IPV6_IFACE_COUNT && MAX_IPV6_IFACE_COUNT > 0)) {
+		NET_WARN("You have %d %s interfaces configured but there "
+			 "are %d network interfaces in the system.",
+			 MAX_IPV6_IFACE_COUNT, "IPv6", iface_count);
+	}
+
+	ARRAY_FOR_EACH(v6_ctx, i) {
+		iface = net_if_get_by_index(i + 1);
+		if (iface == NULL) {
+			continue;
+		}
+
+		v6_ctx[i].iface = iface;
+		v6_ctx[i].dispatcher.local_addr.sa_family = AF_INET6;
+		v6_ctx[i].dispatcher.svc = &v6_svc;
+
+		k_work_init_delayable(&v6_ctx[i].probe_timer, probing);
+
+		ok++;
+	}
+#endif /* CONFIG_NET_IPV6 */
+
+#if defined(CONFIG_NET_IPV4)
+	if ((iface_count > MAX_IPV4_IFACE_COUNT && MAX_IPV4_IFACE_COUNT > 0)) {
+		NET_WARN("You have %d %s interfaces configured but there "
+			 "are %d network interfaces in the system.",
+			 MAX_IPV4_IFACE_COUNT, "IPv4", iface_count);
+	}
+
+	ARRAY_FOR_EACH(v4_ctx, i) {
+		iface = net_if_get_by_index(i + 1);
+		if (iface == NULL) {
+			continue;
+		}
+
+		v4_ctx[i].iface = iface;
+		v4_ctx[i].dispatcher.local_addr.sa_family = AF_INET;
+		v4_ctx[i].dispatcher.svc = &v4_svc;
+
+		k_work_init_delayable(&v4_ctx[i].probe_timer, probing);
+
+		ok++;
+	}
+#endif /* CONFIG_NET_IPV4 */
+
+	if (!ok) {
```

Comment:
nit really, but aren't such bool checks on int against misra? Maybe we could just check `ok == 0 ` and return negative error here.

Reply:
Indeed, will fix.